### PR TITLE
Update singular strings for threats to use 1

### PR DESCRIPTION
**Pull request by JavonDavis**
The Romanian translators are having trouble with the string formatting before the singular noun, This PR changes that to explicitly use 1. 

To test: Test that with a single threat it behaves as expected in https://github.com/wordpress-mobile/WordPress-Android/pull/14021

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/res/values/strings.xml
Comment (AliSoftware) [inline WordPress/src/main/res/values/strings.xml]: Wording nitpick: I know that the previous code was written in a way that the output was already using `1` (and not `one`) in singular form (since it used via `UiStringText("${fixableThreatIds.size}")` in both code paths), but if I'm not mistaken, at least in English it's better to spell-out numbers when using in that kind of context ([source](https://www.grammarly.com/blog/when-to-spell-out-numbers/)) 🙂 

```suggestion
    <string name="scan_idle_with_threats_description_singular">The scan found one potential threat with %1s. Please review the threat below and take action or tap the fix all button. We are %2$s if you need us.</string>
```
(ditto for the other string below of course)

Comment (JavonDavis) [reply WordPress/src/main/res/values/strings.xml]: I thought about it but saw that using "1" was a common pattern, I'll leave the decision on this up to @ashiagr though 🙂

**Comment by JavonDavis** [comment]
Also flagging that changing this to remove the positional argument, will affect all the other translated languages(hence the current lint failure), excluding Romanian... @ashiagr if we're going to do this it'd be probably be best we rename this key to and then merge the release branch into develop to get new a new set of translations and avoid any accidental breakages... wdyt?

File: WordPress/src/main/res/values/strings.xml
Comment (ashiagr) [reply WordPress/src/main/res/values/strings.xml]: Actually yes, 1 is a common pattern in the code but I think it is fine to use "one" as well if it eases our translators job. I'm fine with replacing "1" with "one".

**Comment by ashiagr** [comment]
Thanks for this PR @JavonDavis  🙏.

> Also flagging that changing this to remove the positional argument, will affect all the other translated languages(hence the current lint failure), excluding Romanian... @ashiagr if we're going to do this it'd be probably be best we rename this key to and then merge the release branch into develop to get new a new set of translations and avoid any accidental breakages... wdyt?

Yes, I agree it is better to rename the key and merge the release branch to develop. Should I go ahead and make this change in this PR?

**Comment by JavonDavis** [comment]
> Yes, I agree it is better to rename the key and merge the release branch to develop. Should I go ahead and make this change in this PR?

Yep that'd be great! 🙂 Thanks Ashita! I'll change this PR from a draft since we're going ahead with the change

**Comment by ashiagr** [comment]
This is done @JavonDavis , included screenshots below. Can I merge it to `release/16.8`?

|scan_idle_threats_description_singular | threat_fix_all_confirmation_message_singular | scan_finished_potential_threats_found_message_singular |
|---|---|---|
|device-2021-03-04-102637 | device-2021-03-04-102702 | device-2021-03-04-103437|

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/builders/ScanStateListItemsBuilder.kt
Comment (AliSoftware) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/builders/ScanStateListItemsBuilder.kt]: nit: now that we have different code paths for calling `getHtmlMesageFromStringFormatResId` anyway, probably not worth having an `if` branch just for the `descriptionTestResId` separately.

i.e. remove the `val descriptionTextResId = …` lines and directly use:
```kt
        val descriptionText = if (threatsCount > 1) {
            htmlMessageUtils
                    .getHtmlMessageFromStringFormatResId(
                            R.string.scan_idle_threats_description_plural,
                            "<b>$threatsCount</b>",
                            "<b>${site.name ?: resourceProvider.getString(R.string.scan_this_site)}</b>",
                            clickableText
                    )
        } else {
            htmlMessageUtils
                    .getHtmlMessageFromStringFormatResId(
                            R.string.scan_idle_threats_description_singular,
                            "<b>${site.name ?: resourceProvider.getString(R.string.scan_this_site)}</b>",
                            clickableText
                    )
        }
```

**Review by AliSoftware — COMMENTED** [review]
Code is looking good; only left a suggestion for code readability.

Not marking the PR as ready to merge yet though because @JavonDavis will need to re-run the scripts which updates the `./fastlane/resources/values/strings.xml` file (the one imported by GlotPress) before merging, in order for the new strings to be picked up and to have a chance to have then translated in time before release day (otherwise, those new strings won't be uploaded to GP… until we merge this PR to develop and then start the next code-freeze from develop for the next version next Monday)

@JavonDavis feel free to decide to merge this PR as is anyway… as is as long as you ensure that don't forget to update the file separately during the wrangling of the beta

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/builders/ScanStateListItemsBuilder.kt
Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/builders/ScanStateListItemsBuilder.kt]: Nice suggestion @AliSoftware 🌟, applied in 23da7f2.

**Comment by JavonDavis** [comment]
Thanks @AliSoftware  @ashiagr ! Can I get a 👍  on this? I'll go ahead and update strings.xml when wrangling the new beta and merging into the develop

**Review by AliSoftware — APPROVED** [review]

**Pull request merged**
